backend/src/modules/authentication/core/domain/UserAccount.ts

```diff
@@ -1,29 +1,36 @@
 import { DomainCommandResult } from '../../../../shared/core/domain/DomainCommandResult';
 import { PasswordWasSet } from './event/PasswordWasSet';
+import jwt from 'jsonwebtoken';
+import bcrypt from 'bcrypt';
+import { TokenGenerated } from './event/TokenGenerated';
 
 export class UserAccount {
-  readonly email: string;
-  readonly password: string;
+  readonly userId: string;
+  readonly email: string | undefined;
```

Comment:
Może być bez emaila?

Reply:
hmmm na pewno po coś to zrobiłem, ale nie pamiętam po co....  tak czy inaczej chyba faktycznie nie jest to potrzebne. Rozumiem, że password pozostaje jako string | undefined, bo repo użytkowników będzie jedno, niezależnie w jaki sposób się będą logować / rejestrować, Si? ...więc w przypadku logowania za pomocą googla hasło będzie nam nieznane.

Reply:
Jak dla mnie powinny byc dwa modele usera - jeden dla naszej authentication drugi dla Google. 
Zeby nie bylo wlasnie czegos takiego jak password: string | undefined. 
Chyba to tlumaczylem dosc dlugo pod koniec projektu :P 

Teraz mamy schizofrenie - jestem przez Google czy nie? Nie wiemy tego po typach.